fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeBuiltinConstants.java

```diff
@@ -39,4 +40,16 @@ public class PrivilegeBuiltinConstants {
 
     public static final Set<Long> IMMUTABLE_BUILT_IN_ROLE_IDS = new HashSet<>(Arrays.asList(
             ROOT_ROLE_ID, DB_ADMIN_ROLE_ID, CLUSTER_ADMIN_ROLE_ID, USER_ADMIN_ROLE_ID));
+
+    public static final long ALL_CATALOGS_ID = -1; // -1 represent all catalogs
+
+    public static final String ALL_DATABASES_UUID = "ALL_DATABASES_UUID"; // represent all databases
+    public static final long ALL_DATABASE_ID = -2; // -2 represent all databases
+
+    public static final String ALL_TABLES_UUID = "ALL_TABLES_UUID"; // represent all tables
+
+    public static final long ALL_FUNCTIONS_ID = -3; // -3 represent all functions
+    public static final long GLOBAL_FUNCTION_DEFAULT_DATABASE_ID = -4; // global function default database id
```

Comment:
this will put global function before fuzzy matching non-global function, and we have "short circuit" logic when checking, this may cause wrong checking result.
![image](https://user-images.githubusercontent.com/107105845/226317539-c97ee5c7-1613-4f9c-add0-2f049c06ea89.png)

Reply:
we can modify the compareTo api in FunctionPEntryObject